src/common/crypto/platform/openssl/crypto.cpp

```diff
@@ -665,19 +664,25 @@ static expected::ExpectedBytes TryASN1EncodeMenderCustomBinaryECFormat(
 	r.release();
 	s.release();
 
-	/* Allocate some array guaranteed to hold the DER-encoded structure */
-	vector<uint8_t> der_encoded_byte_array(mender_decode_buf_size);
-	unsigned char *arr_p = &der_encoded_byte_array[0];
-	int len = i2d_ECDSA_SIG(ecSig.get(), &arr_p);
+	/* Get the expected length in bytes of the DER encoded signature */
+	int len = i2d_ECDSA_SIG(ecSig.get(), NULL);
```

Comment:
According to the docs, this allocates a buffer and writes the DER data into it. We work with that buffer and `free()` it in the end.

Reply:
Oh, scratch that. That's the case if the given pointer points to `NULL`, of course. So that would be an alternative solution, but not better in any way, I think.